circuit/src/commonMain/kotlin/com/slack/circuit/BackHandler.kt

```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2023 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit
+
+import androidx.compose.runtime.Composable
+import com.slack.circuit.backstack.BackStack
+
+@Composable
+internal expect fun BackHandler(
+  navigator: Navigator,
+  backstack: BackStack<*>,
+  content: @Composable () -> Unit,
```

Comment:
I find this API confusing 🤔. I think it's fine to just make this an implementation detail on each platform and not have a common one, since they will work in diff ways.

Maybe what we want is an `internal expect NavigableCircuitContent` + `internal BasicNavigableCircuitContent`, where the `internal actual NavigableCircuitContent` functions each handle back handling in a platform-specific way. This would also give you access to a `Modifier` that you could use rather than a wrapper `Box`